zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java

```
@@ -348,6 +349,16 @@ private PendingIncidentUpdateBatch createPendingIncidentBatch(
     for (final var hit : hits) {
       final var entity = hit.source();
       final var newState = IncidentState.createFrom(entity.intent());
+
+      if (newState == null) {
+        final var errMsg =
+            String.format(
+                "Pending incident has a new state of [%s], which is not a valid IncidentState, should be on of %s",
+                entity.intent(), Arrays.toString(IncidentState.values()));
+        logger.error(errMsg);
+        throw new IllegalStateException(errMsg);
```

_Ideally_, you should either throw the exception or log, not both. We should really log where we are handling the exception, rather than where its thrown..